### PR TITLE
add ApiPropertyOptional

### PR DESCRIPTION
**Pull request by tahamansoor**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Docs
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: content/openapi/types-and-parameters.md [hunk 1]
```diff
@@ -34,6 +34,16 @@ export class CreateCatDto {
...
 > info **Hint** Instead of manually annotating each property, consider using the Swagger plugin (see [Plugin](/openapi/cli-plugin) section) which will automatically provide this for you.
 
+use `@ApiPropertyOptional()` for Optional Parameters
+
+```typescript
+import { ApiPropertyOptional} from '@nestjs/swagger';
+
+export class CreateCatDto {
+  @ApiPropertyOptional()
+  breed: string;
```
Comment (micalevisk) [inline content/openapi/types-and-parameters.md:44]:
```suggestion
  breed?: string;
```

**Comment by micalevisk** [comment]
What if you add a new optional field on the `CreateCatDto` that already exists? It will be easier to follow I guess

**Comment by tahamansoor** [comment]
> What if you add a new optional field on the `CreateCatDto` that already exists? It will be easier to follow I guess

` name?: string;` 
are you saying like mentioned above? if yes that does not work I have tried this before

**Comment by micalevisk** [comment]
nvm

I think if you just change the line

image

to

```
As you can see, the definition is empty although the class has a few declared properties.
In order to make the class properties visible to the `SwaggerModule`, we have to either annotate them with the `@ApiProperty()` decorator (or `@ApiOptionalProperty()` for optional fields) or use the CLI plugin (read more in the Plugin section) which will do it automatically:
```

would be enough

**Comment by tahamansoor** [comment]
> nvm
> 
> I think if you just change the line
> 
> image
> 
> to
> 
> ```
> As you can see, the definition is empty although the class has a few declared properties.
> In order to make the class properties visible to the `SwaggerModule`, we have to either annotate them with the `@ApiProperty()` decorator (or `@ApiOptionalProperty()` for optional fields) or use the CLI plugin (read more in the Plugin section) which will do it automatically:
> ```
> 
> would be enough

yes that's a good point

File: content/openapi/types-and-parameters.md (hunk 1 above)
Comment (johannesschobel) [inline content/openapi/types-and-parameters.md:44]: i guess we should use the `?` here?!

```suggestion
  breed?: string;
```

**Review by johannesschobel — CHANGES_REQUESTED** [review]

**Comment by kamilmysliwiec** [comment]
We decided not to mention `@ApiPropertyOptional` and encourage developers to use `@ApiProperty({ required: false })` instead

File: content/openapi/types-and-parameters.md (hunk 1 above)
Comment (tahamansoor) [reply content/openapi/types-and-parameters.md:44]: I have tried it 
It doesn't work

**Pull request closed without merging**